### PR TITLE
Add function to get point on shower axis in altaz

### PR DESCRIPTION
**Pull request by maxnoe**
This can be used to compute the shower axis in telescope / camera frame, e.g. like this:

```python
import astropy.units as u
from astropy.coordinates import AltAz
import matplotlib.pyplot as plt

from ctapipe.calib import CameraCalibrator
from ctapipe.io import EventSource
from ctapipe.coordinates import get_point_on_shower_axis, TelescopeFrame
from ctapipe.visualization import CameraDisplay


plt.style.use("dark_background")
plt.rcParams["savefig.facecolor"] = "0.15"

path = "dataset://lst_prod3_calibration_and_mcphotons.simtel.zst"

if __name__ == "__main__":
    with EventSource(path, focal_length_choice="EQUIVALENT") as source:
        it = iter(source)
        for _ in range(2):
            e = next(it)
        subarray = source.subarray

    calib = CameraCalibrator(subarray)
    calib(e)

    shower = e.simulation.shower
    distance = 5 * u.km

    fig, axs = plt.subplots(2, 2, layout="constrained", figsize=(8, 8))

    telescope_positions = source.subarray.tel_coords
    axis_point = get_point_on_shower_axis(
        core_x=shower.core_x,
        core_y=shower.core_y,
        alt=shower.alt,
        az=shower.az,
        telescope_position=telescope_positions,
        distance=distance,
    )

    for ax, (tel_id, dl1) in zip(axs.flat, e.dl1.tel.items()):
        tel_index = source.subarray.tel_ids_to_indices(tel_id)[0]

        pointing = AltAz(
            alt=e.pointing.tel[tel_id].altitude, az=e.pointing.tel[tel_id].azimuth
        )
        tel_frame = TelescopeFrame(telescope_pointing=pointing)

        geometry = source.subarray.tel[tel_id].camera.geometry
        disp = CameraDisplay(geometry.transform_to(tel_frame), image=dl1.image, ax=ax)

        ax.set_title(f"LST-{tel_id}")
        ax.set_axis_off()
        ax.set_ylabel("")
        ax.set_xlabel("")

        source_tel_frame = AltAz(alt=shower.alt, az=shower.az).transform_to(tel_frame)
        p = axis_point[tel_index].transform_to(tel_frame)

        disp.axes.plot(
            [source_tel_frame.fov_lon.deg, p.fov_lon.deg],
            [source_tel_frame.fov_lat.deg, p.fov_lat.deg],
            color="xkcd:baby blue",
            marker=".",
        )
        disp.overlay_coordinate(source_tel_frame, color="xkcd:light yellow")

    fig.suptitle(
        f"$E={{}}${shower.energy.to(u.GeV):.0f}, Impact = ({shower.core_x:.0f}, {shower.core_y:.0f})"
    )

    plt.savefig("shower_axis.png", dpi=200)
```

shower_axis

### CODE REVIEW

**Review by moralejo — APPROVED** [review]
Looks good to me. Can you show an example with larger images?

**Comment by maxnoe** [comment]
> Looks good to me. Can you show an example with larger images?

Here is a 5 TeV shower:

File: src/ctapipe/coordinates/ground_frames.py [hunk 1]
```diff
@@ -46,11 +46,15 @@ def _earthlocation_to_altaz(location, reference_location):
...
-    """Ground coordinate frame.  The ground coordinate frame is a simple
-    cartesian frame describing the 3 dimensional position of objects
-    compared to the array ground level in relation to the nomial
-    centre of the array.  Typically this frame will be used for
-    describing the position on telescopes and equipment.
+    """Ground coordinate frame.
+
+    The ground coordinate frame is a simple cartesian frame describing the
+    3 dimensional position of objects compared to the array ground level
+    in relation to the nomial center of the array.
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply src/ctapipe/coordinates/ground_frames.py:53]: thx, fixed.

**Comment by moralejo** [comment]
> > Looks good to me. Can you show an example with larger images?
> 
> Here is a 5 TeV shower:
> 
> impact_5tev

Thanks, the lower-E one was not so convincing because the pixelation on small images makes the effect of a misalignment, but this one looks very convincing.

File: src/ctapipe/coordinates/utils.py [hunk 2]
```diff
@@ -62,13 +62,13 @@ def get_point_on_shower_axis(core_x, core_y, alt, az, telescope_position, distan
         Azimuth of primary
     telescope_position : GroundFrame
         Telescope position
-    slant_distance : u.Quantity[length]
+    distance : u.Quantity[length]
         Distance from along the shower axis from the ground of the point returned.
```
Comment (moralejo) [inline src/ctapipe/coordinates/utils.py:66]: from along => along

**Review by kosack — APPROVED** [review]

**Review by moralejo — APPROVED** [review]

**Pull request merged**